### PR TITLE
Fix shellcheck lint errors in Kubemark scripts

### PR DESCRIPTION
**Pull request by mrbobbytables**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**: 
Ensures all kubemark scripts pass hack/verify-shellcheck.sh

**Which issue(s) this PR fixes**:
Working towards #72956

**Special notes for your reviewer**: There are a few parts that should be overhauled in the iks portion of the kubemark script. I have opened #73743 and spoken with @spzala about it.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: test/kubemark/skeleton/util.sh
Comment (mrbobbytables) [inline test/kubemark/skeleton/util.sh]: This mimics the color coding in [hack/lib/util.sh](https://github.com/kubernetes/kubernetes/blob/master/hack/lib/util.sh#L778-L787) and is sourced by both [test/kubemark/start-kubemark.sh](https://github.com/kubernetes/kubernetes/blob/master/test/kubemark/start-kubemark.sh#L26) and [test/kubemark/stop-kubemark.sh](https://github.com/kubernetes/kubernetes/blob/master/test/kubemark/stop-kubemark.sh#L21).

File: cluster/kubemark/gce/config-default.sh
Comment (mrbobbytables) [inline cluster/kubemark/gce/config-default.sh]: The majority of variables in this script come up as [SC2034](https://github.com/koalaman/shellcheck/wiki/SC2034) or _"foo appears unused. Verify it or export it." _

With that I think it's safe to just disable the check for the entire script.

File: test/kubemark/iks/util.sh
Comment (mrbobbytables) [inline test/kubemark/iks/util.sh]: Follow-up issue regarding clean up: #73743

File: cluster/kubemark/gce/config-default.sh
Comment (BenTheElder) [reply cluster/kubemark/gce/config-default.sh]: yeah this is a wonderful "config script" that configures another script by exporting envs. 🙃

File: test/kubemark/iks/util.sh
Comment (BenTheElder) [inline test/kubemark/iks/util.sh]: what was the lint here? :O

Comment (mrbobbytables) [reply test/kubemark/iks/util.sh]: [SC2091](https://github.com/koalaman/shellcheck/wiki/SC2091) - _Remove surrounding $() to avoid executing output._

I believe the intent here is to execute it. `bx cs cluster-config kubeMasterTester --admin` outputs something akin to:
 ```
export KUBECONFIG=/Users/<user_name>/.bluemix/plugins/container-service/clusters/pr_firm_cluster/kube-config-prod-par02-pr_firm_cluster.yml`
```
See step 3 here: https://developer.ibm.com/recipes/tutorials/deploy-a-kubernetes-cluster-in-ibm-cloud-container-service#Setting-up-the-Cluster-Environment.

It's sort of like `eval $(minikube docker-env)`

Can swap from eval to storing it as a string and executing it that way if it'd be preferred.

Comment (BenTheElder) [reply test/kubemark/iks/util.sh]: ah, this makes sense as is. thanks!

**Review by BenTheElder — APPROVED** [review]
/lgtm
thanks!

**Comment by BenTheElder** [comment]
/approve
will need other approvers for directories besides test/

**Comment by mrbobbytables** [comment]
Thanks @BenTheElder!
/assign @gmarek

**Comment by mrbobbytables** [comment]
rebased after #74193 fixed `./test/kubemark/pre-existing/util.sh`

File: test/kubemark/iks/util.sh
Comment (spiffxp) [inline test/kubemark/iks/util.sh]: Should this instead be exported, and the function commented as such?  I think this is how I've seen this handled elsewhere

File: test/kubemark/iks/util.sh
Comment (spiffxp) [inline test/kubemark/iks/util.sh]: `head -n 1`?

File: test/kubemark/resources/start-kubemark-master.sh
Comment (spiffxp) [inline test/kubemark/resources/start-kubemark-master.sh]: whitespace, I nearly missed this
```suggestion
    local files=""
```

File: test/kubemark/resources/start-kubemark-master.sh
Comment (spiffxp) [inline test/kubemark/resources/start-kubemark-master.sh]: nit: if SC2155 is going to force us to break up decls, I'd prefer to see this right next the the decl

File: test/kubemark/iks/util.sh
Comment (mrbobbytables) [inline test/kubemark/iks/util.sh]: For reference [SC2162](https://github.com/koalaman/shellcheck/wiki/SC2162) - _"read without -r will mangle backslashes"_

File: test/kubemark/run-e2e-tests.sh
Comment (spiffxp) [inline test/kubemark/run-e2e-tests.sh]: You use `ARGS[*]` in the `go run` above, but `ARGS[@]` here, is that on purpose?

**Review by spiffxp — COMMENTED** [review]
a few nits/questions

File: test/kubemark/run-e2e-tests.sh
Comment (mrbobbytables) [reply test/kubemark/run-e2e-tests.sh]: Yes, my thought was the original was it being interpreted as a string since it was a part of:
```
--test_args="--e2e-verify-service-account=false --dump-logs-on-failure=false ${ARGS}"
```

In the 2nd, it was additional arguments being passed and should the be independent quoted arguments.

File: test/kubemark/resources/start-kubemark-master.sh
Comment (mrbobbytables) [reply test/kubemark/resources/start-kubemark-master.sh]: Yeah, it presented itself fine in vscode, noticed it too when it was pushed up -_- fixed it in vim and repushed.

File: test/kubemark/resources/start-kubemark-master.sh
Comment (mrbobbytables) [reply test/kubemark/resources/start-kubemark-master.sh]: Can do that.

File: test/kubemark/iks/util.sh
Comment (mrbobbytables) [reply test/kubemark/iks/util.sh]: From what I had originally gathered, `CA_CERT_BASE64` and eventual generated kubeconfig was expecting a single ca pem file to be read and encoded. The large glob used (`*ca*`) could potentially pick up multiple results. The `head -n 1` was a means to limit it to always return a single result. However there is potential for it to return the wrong ca or match other files. In either case a failure would occur, but the limit at least ensures its always getting a single result.

File: test/kubemark/iks/util.sh
Comment (mrbobbytables) [reply test/kubemark/iks/util.sh]: I think it'd be okay, if so there are 2 others that should also be exported. `KUBECTL` and `TEST_CLUSTER_API_CONTENT_TYPE`. Both are set within a function in the util script, but used elsewhere.

Comment (spiffxp) [reply test/kubemark/iks/util.sh]: Yeah that sounds good

**Comment by spiffxp** [comment]
/approve
/hold
a few things to resolve
... and also needs-rebase

File: test/kubemark/resources/start-kubemark-master.sh
Comment (mrbobbytables) [reply test/kubemark/resources/start-kubemark-master.sh]: Done.

File: test/kubemark/iks/util.sh
Comment (mrbobbytables) [reply test/kubemark/iks/util.sh]: Should be good. :)

File: cluster/kubemark/iks/config-default.sh
Comment (BenTheElder) [inline cluster/kubemark/iks/config-default.sh]: tiny nit: double `#` ?

File: hack/.shellcheck_failures
Comment (BenTheElder) [inline hack/.shellcheck_failures]: this diff is weird, what happened here?

Comment (BenTheElder) [reply hack/.shellcheck_failures]: ah removed in the next commit

Comment (BenTheElder) [reply hack/.shellcheck_failures]: seems like the next commit should get squashed into this one, as otherwise it's a no-op?

File: test/kubemark/iks/util.sh
Comment (BenTheElder) [reply test/kubemark/iks/util.sh]: still needs the comment I think?

File: test/kubemark/resources/start-kubemark-master.sh
Comment (BenTheElder) [inline test/kubemark/resources/start-kubemark-master.sh]: you can declare `local files` without assigning, which is probably more clear

**Comment by BenTheElder** [comment]
thanks for fixing these, couple of minor comments 👍

File: hack/.shellcheck_failures
Comment (mrbobbytables) [reply hack/.shellcheck_failures]: That likely occurred while I was rebasing the first time around, I'll go through and make sure everything is good.

File: test/kubemark/iks/util.sh
Comment (mrbobbytables) [reply test/kubemark/iks/util.sh]: They do -_- I completely forgot to add that part. Will resolve shortly.

File: test/kubemark/iks/util.sh
Comment (spiffxp) [inline test/kubemark/iks/util.sh]: As you said above, you'll want to add a comment about this and `TEST_CLUSTER_API_CONTENT_TYPE` to `complete-login`

**Comment by spiffxp** [comment]
very close

**Comment by mrbobbytables** [comment]
Thanks @spiffxp!

/pony thankyou

**Comment by mrbobbytables** [comment]
bah XD

/pony thanks

**Pull request merged**
